Changes needed to get watch to build on Solaris.

This patch should be submitted upstream.

--- procps-ng-3.3.12/watch.c.orig	2017-02-23 09:00:02.567228629 +0000
+++ procps-ng-3.3.12/watch.c	2017-03-07 13:38:42.017966296 +0000
@@ -372,7 +372,8 @@
 	char *ts = ctime(&t);
 	char *header;
 	char *right_header;
-	char hostname[HOST_NAME_MAX + 1];
+	int max_host_name_len = (int) sysconf(_SC_HOST_NAME_MAX);
+	char hostname[max_host_name_len + 1];
 	int command_columns = 0;	/* not including final \0 */
 
 	gethostname(hostname, sizeof(hostname));
